pkg/host/types/interfaces.go

```diff
@@ -138,7 +140,7 @@ type SriovInterface interface {
 	// GetVfInfo returns the virtual function information is the operator struct from the host information
 	GetVfInfo(pciAddr string, devices []*ghw.PCIDevice) sriovnetworkv1.VirtualFunction
 	// SetVfGUID sets the GUID for a virtual function
-	SetVfGUID(vfAddr string, pfLink netlink.Link) error
+	SetVfGUID(vfAddr string, guid net.HardwareAddr, pfLink netlink.Link) error
```

Comment:
Let's remove this method from the `SriovInterface` and move it to a new `InfinibandInterface`. 
+ let's remove guid arg from the function, I think guid generation should happen inside the function